microceph/api/client_configs.go

```diff
@@ -0,0 +1,180 @@
+package api
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microceph/microceph/client"
+	"github.com/canonical/microceph/microceph/common"
+	"github.com/canonical/microceph/microceph/database"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+)
+
+// Top level client API
+var clientCmd = rest.Endpoint{
+	Path: "client",
+	Get:  rest.EndpointAction{Handler: cmdClientGet, ProxyTarget: true},
+}
+
+func cmdClientGet(s *state.State, r *http.Request) response.Response {
+	return response.EmptySyncResponse
+}
+
+// client configs API
+var clientConfigsCmd = rest.Endpoint{
+	Path: "client/configs",
+	Put:  rest.EndpointAction{Handler: cmdClientConfigsPut, ProxyTarget: true},
+	Get:  rest.EndpointAction{Handler: cmdClientConfigsGet, ProxyTarget: true},
+}
+
+func cmdClientConfigsGet(s *state.State, r *http.Request) response.Response {
+	var req types.ClientConfig
+	var configs database.ClientConfigItems
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.InternalError(err)
+	}
+
+	if req.Host == common.ClientConfigGlobalHostConst {
+		configs, err = database.ClientConfigQuery.GetAll(s)
+	} else {
+		configs, err = database.ClientConfigQuery.GetAllForHost(s, req.Host)
+	}
+	if err != nil {
+		logger.Errorf("failed fetching client configs: %v for %v", err, req)
+		return response.SyncResponse(false, nil)
+	}
+
+	logger.Infof("Database Response: %v", configs)
+
+	return response.SyncResponse(true, configs.GetClientConfigSlice())
+}
+
+// Implements the render .conf file at that particular host.
+func cmdClientConfigsPut(s *state.State, r *http.Request) response.Response {
+	// Check if microceph is bootstrapped.
+	err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		isFsid, err := database.ConfigItemExists(ctx, tx, "fsid")
+		if err != nil || !isFsid {
+			return fmt.Errorf("cluster is not bootstrapped yet: %v", err)
```

Comment:
if `err` is nil (because there's no fsid), users are going to get a rather confusing message here. I suggest displaying a different error message for that case.

Reply:
good catch, changing.